src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanAgentList.hpp

```diff
@@ -49,12 +49,20 @@ class TScanAgentList final : public AbstractHandler<std::shared_ptr<TScanContext
         static SocketDBWrapper wdbSocketWrapper(WDB_SOCKET);
 
         nlohmann::json response;
-        wdbSocketWrapper.query(WazuhDBQueryBuilder::builder().agentGetOsInfoCommand(agent.id).build(), response);
 
-        // Validate the response
-        if (response.empty())
+        try
         {
-            logWarn(WM_VULNSCAN_LOGTAG, "Empty response for agent '%s' in Wazuh-DB 'sys_' query", agent.id.c_str());
+            // Execute query
+            wdbSocketWrapper.query(WazuhDBQueryBuilder::builder().agentGetOsInfoCommand(agent.id).build(), response);
```

Comment:
You are changing the behavior by catching the exception here, since the idea is to abort the re-scan and turn it to delayed when one of the two types (os and packages) fail

Reply:
add test to avoid this type of change in the future please.

Reply:
Reverted!